src/Homepage/Homepage.js

```diff
@@ -1,13 +1,22 @@
-
 import './Homepage.css';
 
 
-export const Homepage = () => {
+export const Homepage = (props) => {
   return (
     <div className="homepage">
       <div className="character-select">
-        <button className="generator">Generator</button>
-        <button className="guesser">Guesser</button>
+        <button 
+        className="generator"
+        onClick={
+          props.designateRole(false)
+        }
+        >Generator</button>
+        <button 
+        className="guesser"
+        onClick={
+          props.designateRole(true)
+        }
```

Comment:
This whole file is sexy. 

Functional component, propping down a callback that calls a fetch... woof, I need a drink